@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -2221,27 +2233,50 @@ export class TextElementsRenderer {
             }
             screenPoints.push(tempScreenPosition.clone());
 
-            if (screenPoint.x < minX) {
-                minX = screenPoint.x;
-            }
-            if (screenPoint.x > maxX) {
-                maxX = screenPoint.x;
+            const i = screenPoints.length - 1;
+
+            if (indexOfFirstVisibleScreenPoint < 0) {
+                if (this.m_screenCollisions.screenBounds.contains(screenPoint.x, screenPoint.y)) {
+                    indexOfFirstVisibleScreenPoint = i;
+                }
             }
-            if (screenPoint.y < minY) {
-                minY = screenPoint.y;
+
+            if (i < 1) {
+                continue;
             }
-            if (screenPoint.y > maxY) {
-                maxY = screenPoint.y;
+
+            const pointA = screenPoint;
+            const pointB = screenPoints[i - 1];
+
+            const nextDeltaX = pointA.x - pointB.x;
+            const nextDeltaY = pointA.y - pointB.y;
+            const nextLength = Math.sqrt(nextDeltaX * nextDeltaX + nextDeltaY * nextDeltaY);
+            const nextLenInverse = 1 / nextLength;
+            const nextNormalX = nextDeltaX * nextLenInverse;
+            const nextNormalY = nextDeltaY * nextLenInverse;
```

Comment:
This could be nicely put into the utilities as a function "calculateLengthAndNormalFromPoints()" or something like that.